win32ss/gdi/ntgdi/freetype.c

```diff
@@ -6795,8 +6795,9 @@ IntExtTextOutW(
     FONT_CACHE_ENTRY Cache;
     FT_Matrix mat;
     BOOL bNoTransform;
-    DWORD ch0, ch1;
+    DWORD ch0, ch1, etx = 3; // etx is End of Text
```

Comment:
I have been running with this PR on my local install and found that the ETX is often used and it has a bitSize.cx of zero.
This seems OK to me since it is only a marker, so I wanted to exclude it from being recorded in the debug log.

Reply:
Why the value "3" and not any other? :thinking:

Reply:
@HBelusca, The ASCII value of 3 has been known as an "End of Text" marker from the days of the Teletype.
See https://simple.m.wikipedia.org/wiki/File:ASCII-Table-wide.svg and any other complete ASCII reference.
It seems that this choice has also been used within Abiword to indicate the same condition.
This is the internal representation of what today has become mostly <CR> and <CR><LF>.

Reply:
As a casual observer, I did not make the connection that this was just an ASCII code. If your comment said "etx is ASCII End of Text" it would be clearer.

Reply:
@whindsaks, Thanks for your comments. I added text in accordance with your suggestion. I am a bit 'old school' and the 'etx' is somewhat obvious to me.

Reply:
Then I wonder in which situations / which programs you have encountered that use an end-of-text byte somewhere in the string buffer...
(Regarding CR/LF, they have of course different values.)

Reply:
@HBelusca, I believe there are many examples and some are included in our ReactOS regression tests.
If you go to win32ss/gdi/ntgdi/freetype.c and change the two lines at 7067 to the following:

7067: if (ch0 != L' ' && /* ch0 != etx && */ bitSize.cx == 0) // Remove check for ch0 != etx
7068:    DPRINT1("WARNING: WChar 0x%04x has a bitSize.cx of zero\n", ch0);

You will see lines such as this appear when you run a full 'rosautotest'
(win32ss/gdi/ntgdi/freetype.c:7068) WARNING: WChar 0x0003 has a bitSize.cx of zero

These appear in the following tests which indicate that they are used there:

1. Module: mshtml, Test: htmldoc
2. Module: mshtml, Test: script
3. Module: riched20, Test: editor
4. Module: riched32, Test: editor
5. Module: usp10, Test: usp10

So I believe that these are fairly commonly used on the functions referenced above. Thanks.